### PR TITLE
Generate index file

### PR DESCRIPTION
**Pull request by Mathieudewet**
WIP (Cleaning).

### CODE REVIEW

File: bin/generate-index [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env node
+
+const { docTitle } = require('../constants');
+const chapterHelper = require('../src/helper/chapterHelper');
```
Comment (Mathieudewet) [inline bin/generate-index]: { processChapter } = require('../src/helper/chapterHelper');

File: bin/generate-index [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const { readFileSync, writeFile } = require('fs');
+const { resolve } = require('path');
+const jsyaml = require('js-yaml');
+const fileProcessor = require('./fileProcessor');
+
+const loadedOutline = jsyaml.safeLoad(readFileSync(`${__dirname}/../config/outline.yaml`, 'utf8'));
+
+const index = `# ${docTitle}\n\n${loadedOutline.map((chapter) => chapterHelper.processChapter(chapter, fileProcessor)).join('\n\n')}`;
```
Comment (Mathieudewet) [inline bin/generate-index]: `chapterHelper.processChapter` => `processChapter`

File: src/helper/chapterHelper.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  processItemAnchor(anchor, path, index, depth) {
+    return `${'  '.repeat(depth)}${index + 1}.${(index + 1).toString().length !== 2 ? '  ' : ' '}[${anchor.title}](${path}#${anchor.id})\n${anchor.anchors ? this.processItemAnchors(anchor, path , ++depth) : ''}`;
+  },
+  processChapter(chapter, fileProcessor) {
+    return `## ${chapter.title}\n\n${this.processChapterItems(chapter, fileProcessor)}`;
+  },
+};
+
+module.exports = chapterHelper;
```
Comment (dunglas) [inline src/helper/chapterHelper.js]: Can't you use ES6 modules instead?

File: bin/generate-index [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+writeFile(resolve(__dirname, '../src/pages/docs/index.md'), index, error => error && console.warn(error));
```
Comment (dunglas) [inline bin/generate-index]: Is it really necessary to generate a Markdown file? Can't we use directly the existing YAML file as a source for Gatsby?

https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-yaml
https://github.com/gatsbyjs/gatsby/issues/3195

**Pull request closed without merging**